service/matching/taskReader.go

```diff
@@ -133,11 +133,24 @@ dispatchLoop:
 			}
 			for {
 				// find isolation group of the task
-				isolationGroup := tr.getIsolationGroupForTask(tr.cancelCtx, taskInfo)
+				isolationGroup, err := tr.getIsolationGroupForTask(tr.cancelCtx, taskInfo)
+				if err != nil {
+					// it only errors when the tasklist is a sticky tasklist and
+					// the sticky pollers are not available, in this case, we just complete the task
+					// and let the decision get timed out and rescheduled to non-sticky tasklist
+					if err == _stickyPollerUnavailableError {
+						tr.completeTask(taskInfo, nil)
```

Comment:
I might need help understanding what the consequences of completing a task like this are

Reply:
I'm also still slightly unclear why this is in a loop

Reply:
outer loop is for draining all tasks from the buffer
inner loop is for retrying on dispatching one task to a poller